script.js

```diff
@@ -168,34 +159,42 @@ var Phonebook = {
     "Nelson Tan"    :   63396565
 };
 
+var promptKeys = Object.keys(Phonebook).join(", ");
+var userPromptOneName = prompt("Please Enter One of the Names: " + promptKeys);
 
+if (Phonebook[userPromptOneName]) {
+  console.log(userPromptOneName + "'s phone number is: " + Phonebook[userPromptOneName]);
+} else {
+  console.log("Invalid name");
+};
 
+var userPromptTwoName = prompt("Please Enter One of the Names: " + promptKeys);
+var userPromptTwoNumber = prompt("Enter the person's new phone number");
 
+if (Phonebook[userPromptTwoName]) {
+  Phonebook[userPromptTwoName] = userPromptTwoNumber;
+};
 
+var userPromptThreeName = prompt("Please Enter One of the Names: " + promptKeys + " or a New Name");
+var userPromptThreeNumber = prompt("Enter a new phone number");
 
+if (!Phonebook[userPromptThreeName]) {
+  Phonebook[userPromptThreeName] = userPromptThreeNumber;
+};
```

Comment:
Just something to keep in mind, although we will likely not be going into sufficient depth to see the impact of this, checking for object properties like this is not generally safe. For example, try typing __proto__ into the prompt :) Look into Object methods on MDN, specifically Object.hasOwnProperty()